src/Ninject.Web.AspNetCore.Test/Unit/ServiceProviderTest.cs

```diff
@@ -0,0 +1,147 @@
+		public void OptionalNonExistingSingleServiceResolvedToNull()
+		{
+			StandardKernel kernel = CreateTestKernel();
+			var provider = CreateServiceProvider(kernel);
+			
+			provider.GetService(typeof(IWarrior)).Should().BeNull();
+		}
+
+		[Fact]
+		public void OptionalExistingMultipleServicesResolvedQueriedAsList()
+		{
+			StandardKernel kernel = CreateTestKernel();
+			kernel.Bind<IWarrior>().To<Samurai>();
+			kernel.Bind<IWarrior>().ToConstant(new Ninja("test"));
+			var provider = CreateServiceProvider(kernel);
+			
+			var result = provider.GetService(typeof(IList<IWarrior>)) as IEnumerable<IWarrior>;
+			
+			result.Should().NotBeNull();
+			var resultList = result.ToList();
+			resultList.Should().HaveCount(2);
+			resultList.Should().Contain(x => x is Samurai);
+			resultList.Should().Contain(x => x is Ninja);
+		}
+
+		[Fact]
+		public void OptionalExistingMultipleServicesNotResolvedWhenNotQueriedAsList()
+		{
+			StandardKernel kernel = CreateTestKernel();
+			kernel.Bind<IWarrior>().To<Samurai>();
+			kernel.Bind<IWarrior>().ToConstant(new Ninja("test"));
+			var provider = CreateServiceProvider(kernel);
+
+			provider.GetService(typeof(IWarrior)).Should().BeNull();			
+		}
+
+		[Fact]
+		public void RequiredExistingSingleServiceResolved()
+		{
+			StandardKernel kernel = CreateTestKernel();
+			kernel.Bind<IWarrior>().To<Samurai>();
+			var provider = CreateServiceProvider(kernel);
+
+			provider.GetRequiredService(typeof(IWarrior)).Should().NotBeNull().And.BeOfType(typeof(Samurai));
+		}
+
+		[Fact]
+		public void RequiredNonExistingSingleServiceResolvedToException()
+		{
+			StandardKernel kernel = CreateTestKernel();
+			var provider = CreateServiceProvider(kernel);
+
+			Action action = () => provider.GetRequiredService(typeof(IWarrior));
+			action.Should().Throw<ActivationException>().WithMessage("*No matching bindings are available*");
+		}
+
+		[Fact]
+		public void RequiredExistingMultipleServicesResolvedQueriedAsList()
+		{
+			StandardKernel kernel = CreateTestKernel();
+			kernel.Bind<IWarrior>().To<Samurai>();
+			kernel.Bind<IWarrior>().ToConstant(new Ninja("test"));
+			var provider = CreateServiceProvider(kernel);
+
+			var result = provider.GetRequiredService(typeof(IList<IWarrior>)) as IEnumerable<IWarrior>;
+
+			result.Should().NotBeNull();
+			var resultList = result.ToList();
+			resultList.Should().HaveCount(2);
+			resultList.Should().Contain(x => x is Samurai);
+			resultList.Should().Contain(x => x is Ninja);
+		}
+
+		[Fact]
+		public void RequiredExistingMultipleServicesResolvedToExceptionWhenNotQueriedAsList()
+		{
+			StandardKernel kernel = CreateTestKernel();
+			kernel.Bind<IWarrior>().To<Samurai>();
+			kernel.Bind<IWarrior>().ToConstant(new Ninja("test"));
+			var provider = CreateServiceProvider(kernel);
+
+			Action action = () => provider.GetRequiredService(typeof(IWarrior));
+			action.Should().Throw<ActivationException>().WithMessage("*More than one matching bindings are availabl*");
```

Comment:
If we don't implement ISupportRequiredService we would just get a InvalidOperationException without any nice explanation like in Ninject. Here Ninject lists all the found matches and explains why it doesn't work whereas the standard implementation just say nothing found.